doc/engineering/coding_standards.md

```diff
@@ -0,0 +1,236 @@
+
+For quick reference, here are the relevant git commands:
+
+```shell
+git config --global user.name "Your Name Comes Here"
+git config --global user.email you@yourdomain.example.com
+```
+
+Make separate commits for separate changes. If you cannot describe what the commit does in one sentence, it is probably a mix of changes and should be separated into several commits.
+
+A well-crafted git commit message is the best way to communicate context about a change to fellow developers (and indeed to their future selves). A diff will tell you what changed, but only the commit message can properly tell you why.
+
+### General Guidelines
+
+Here are some general guidelines for submitting new code.
+
+* Follow the guidelines for writing good commit messages.
+* Do not commit commented-out code or files that are no longer needed.
+* Check for unnecessary whitespace before committing with git diff --check.
+* All pull requests should be issued to resolve a Jira ticket.  The branch name of the submission should be Jira ticket ID, such as KAZOO-42.
+* Ensure that each pull request addresses only the work related to the ticket it is resolving.
+* Keep the changes in a pull request as concise as possible.  If the pull request is a bug fix it should be the simplest possible approach with minimal code changes.
+* All work should be rebased to the originating branch prior to pull request submission.
+
+### Bug Fix Guidelines
+
+Bug fixes must never introduce new features.
+
+Bug fix branches must have originated from an updated version branch in the project which matches the affects version of the bug report.  The pull request should be made against the version branch and all subsequent version branches including master.
+
+All non-bug fix work branches must have originated from an updated master branch in the project and the pull request must be issued against the master branch.
+
+If the pull request was not for the master branch then all subsequent version branches must also have a pull request opened with the bug fix.  It is acceptable to work with 2600Hz to organize a sequence of pull requests when more convenient (such as refactoring the pull request against 3.22 and re-opening it for master).
+
+A test should also be submitted to ensure that the bug is not accidentally reintroduced in the future.
+
+### New Functionality/Features Guidelines
+
+If you are implementing a new feature, also write new test cases.  The primary reason for writing test cases is not to prove that the new feature works correctly, but to make sure that it will be noticed if future changes — perhaps to code that seems unrelated — break the feature.
+
+If you are implementing a new feature, also update the documentation to describe the feature.  If the documentation belongs to a missing document, such as an addition to an undocumented API it is acceptable to only document the addition.  It is nice to create the full missing documentation section ;)
```

Comment:
Which 'an' do you have issue with? "an addition" and "an undocumented API" are right.